python/ray/data/_internal/execution/interfaces.py

```diff
@@ -390,6 +420,34 @@ def incremental_resource_usage(self) -> ExecutionResources:
         return ExecutionResources()
 
 
+class OutputIterator(Iterator[RefBundle]):
+    """Iterator used to access the output of an Executor execution.
+
+    This is a blocking iterator, and must be thread-safe (i.e., multiple threads can
```

Comment:
The impl below doesn't seem to thread-safe. Or is it assuming the `base` iterable is thread-safe? If so we'd document that assumption.

Reply:
Yeah, it is more a statement that all our iterators are thread-safe (because they can be used from multiple threads for split / streaming execution).